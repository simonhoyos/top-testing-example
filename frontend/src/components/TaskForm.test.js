import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { TaskForm } from './TaskForm';

describe('TaskForm', () => {
  it('should render correctly', () => {
    const { container } = render(<TaskForm />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should change input value', () => {
    const handleChange = jest.fn();
    const event = {
      target: {
        name: 'name',
        value: 'value',
      },
    };
    const { getByLabelText } = render(<TaskForm handleChange={handleChange} />);

    const input = getByLabelText('Name');

    fireEvent.change(input, event);

    expect(handleChange).toHaveBeenCalled();
  });
})
