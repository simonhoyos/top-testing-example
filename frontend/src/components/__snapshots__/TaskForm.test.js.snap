// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm should render correctly 1`] = `
<form>
  <label
    for="name"
  >
    Name
  </label>
  <input
    id="name"
    name="name"
    type="text"
    value=""
  />
  <button>
    Submit
  </button>
</form>
`;
